### PR TITLE
CORTX-29309:Update consul and jq software version 

### PR DESCRIPTION
**Pull request by pavankrishnat**
Consul version update to 1.11.4 is required for getting some fixes from
hashicorp project, which are required to resolve some issues seen in
cortx project - one such issue is - CORTX-28823.
Jq version update is required for the issue seen - CORTX-29150

### Specification of requested package

Details | Values
------ | ------
Package Name  |  consul 
Version | 1.11.4
license | hashicorp consul
Source  | https://www.consul.io/downloads

Details | Values
------ | ------
Package Name  | jq
Version | 1.6-3
license |
Source  | http://mirror.centos.org/centos/8-stream/AppStream/x86_64/os/Packages/jq-1.6-3.el8.x86_64.rpm

### Please add below details about package

* [x] Purpose of using new SW ?
To resolve some issues seen in cortx project - one such issue is - CORTX-28823.
* [x] Location of the source code of new opensource SW ?
consul:- https://www.consul.io/downloads
jq:- http://mirror.centos.org/centos/8-stream/AppStream/x86_64/os/Packages/jq-1.6-3.el8.x86_64.rpm
* [x] Did you evaluate existing SW and confirm they do not serve the purpose ? If yes, include the comparison. If no, why ?
consul:- https://www.consul.io/docs/k8s/upgrade
* [ ] Who is building it or how it gets built ? If yes, Which component/rpm will it be included ? i.e. name of the rpm which includes SW.
* [x] Download location of opensource software package ?
consul:- https://www.consul.io/downloads
jq:- http://mirror.centos.org/centos/8-stream/AppStream/x86_64/os/Packages/jq-1.6-3.el8.x86_64.rpm
* [ ] Who will be supporting this package ?
* [ ] Is there a configuration required for this? Who will be configuring it ?
* [ ] Which component is responsible for deploying on nodes ?
* [ ] What is the licensing policy ? Is it approved by legal ?

### CODE REVIEW

**Comment by gauravchaudhari02** [comment]
Updated packages tested with custom-ci
https://eos-jenkins.colo.seagate.com/job/GitHub-custom-ci-builds/job/generic/job/custom-ci/5438/
Deployment - https://eos-jenkins.colo.seagate.com/job/Cortx-Automation/job/RGW/job/setup-cortx-rgw-cluster/1316/

**Review by gauravchaudhari02 — APPROVED** [review]

**Comment by vaibhavparatwar** [comment]
@ujjwalpl @johnbent @mssawant @pawankumarsrivastava need your approval here

**Comment by shailesh-vaidya** [comment]
@nitin-seagate @madhavemuri @kupranay @sachinpunadikar @cdeshmukh 
We are planning to update jq and consul package version as part of this change. We have tested image generation and deployment with this change. Refer - https://github.com/Seagate/cortx-re/pull/926#issuecomment-1075474525
Could you please check and confirm this change will not impact the functionality of your component.

**Comment by sachinpunadikar** [comment]
@rahul-tripathi-git please check for utils rpm & approve this PR likewise
Please make sure to have utils PR to make required checks
cc - @selvakumaar5496

**Comment by pavankrishnat** [comment]
@vaibhavparatwar,  I can see consul client are with latest version and hctl drive-state cmd working with latest jq version
```
[root@ssc-vm-rhev4-2195 ~]# kubectl exec -it cortx-data-ssc-vm-rhev4-2194-96c4c4447-tdvqb -c cortx-hax -- /bin/bash
[root@cortx-data-headless-svc-ssc-vm-rhev4-2194 /]# consul members
Node                                         Address              Status  Type    Build   Protocol  DC   Partition  Segment
consul-server-0                              172.18.240.154:8301  alive   server  1.10.0  2         dc1  default    <all>
consul-server-1                              172.18.76.24:8301    alive   server  1.10.0  2         dc1  default    <all>
consul-server-2                              172.18.0.21:8301     alive   server  1.10.0  2         dc1  default    <all>
cortx-data-headless-svc-ssc-vm-rhev4-2193    172.18.76.32:8301    alive   client  1.11.4  2         dc1  default    <default>
cortx-data-headless-svc-ssc-vm-rhev4-2194    172.18.240.156:8301  alive   client  1.11.4  2         dc1  default    <default>
cortx-data-headless-svc-ssc-vm-rhev4-2195    172.18.0.29:8301     alive   client  1.11.4  2         dc1  default    <default>
cortx-server-headless-svc-ssc-vm-rhev4-2193  172.18.76.34:8301    alive   client  1.11.4  2         dc1  default    <default>
cortx-server-headless-svc-ssc-vm-rhev4-2194  172.18.240.158:8301  alive   client  1.11.4  2         dc1  default    <default>
cortx-server-headless-svc-ssc-vm-rhev4-2195  172.18.0.31:8301     alive   client  1.11.4  2         dc1  default    <default>
ssc-vm-rhev4-2193.colo.seagate.com           172.18.76.23:8301    alive   client  1.10.0  2         dc1  default    <default>
ssc-vm-rhev4-2194.colo.seagate.com           172.18.240.149:8301  alive   client  1.10.0  2         dc1  default    <default>
ssc-vm-rhev4-2195.colo.seagate.com           172.18.0.22:8301     alive   client  1.10.0  2         dc1  default    <default>

[root@cortx-data-headless-svc-ssc-vm-rhev4-2194 /]# rpm -qa | grep consul
consul-1.11.4-1.x86_64

[root@cortx-data-headless-svc-ssc-vm-rhev4-2194 /]# yum list | grep jq
Failed to set locale, defaulting to C.UTF-8
jq.x86_64                                              1.6-3.el8                                         @cortx-storage.colo.seagate.com_releases_cortx_github_integration-custom-ci_rockylinux-8.4_custom-build-5438_3rd_party
jq.i686                                                1.5-12.el8      
```        

```
[root@cortx-data-headless-svc-ssc-vm-rhev4-2194 /]# hctl status -d
Byte_count:
    critical_byte_count : 0
    damaged_byte_count : 0
    degraded_byte_count : 0
    healthy_byte_count : 0
Data pool:
    # fid name
    0x6f00000000000001:0x93 'storage-set-1__sns'
Profile:
    # fid name: pool(s)
    0x7000000000000001:0xd2 'Profile_the_pool': 'storage-set-1__sns' 'storage-set-1__dix' None
Services:
    cortx-data-headless-svc-ssc-vm-rhev4-2195
    [started]  hax                 0x7200000000000001:0x2b         inet:tcp:cortx-data-headless-svc-ssc-vm-rhev4-2195@22001
    [started]  ioservice           0x7200000000000001:0x2e         inet:tcp:cortx-data-headless-svc-ssc-vm-rhev4-2195@21001
    [started]  ioservice           0x7200000000000001:0x3b         inet:tcp:cortx-data-headless-svc-ssc-vm-rhev4-2195@21002
    [started]  confd               0x7200000000000001:0x48         inet:tcp:cortx-data-headless-svc-ssc-vm-rhev4-2195@21003
    cortx-data-headless-svc-ssc-vm-rhev4-2194  (RC)
    [started]  hax                 0x7200000000000001:0x7          inet:tcp:cortx-data-headless-svc-ssc-vm-rhev4-2194@22001
    [started]  ioservice           0x7200000000000001:0xa          inet:tcp:cortx-data-headless-svc-ssc-vm-rhev4-2194@21001
    [started]  ioservice           0x7200000000000001:0x17         inet:tcp:cortx-data-headless-svc-ssc-vm-rhev4-2194@21002
    [started]  confd               0x7200000000000001:0x24         inet:tcp:cortx-data-headless-svc-ssc-vm-rhev4-2194@21003
    cortx-data-headless-svc-ssc-vm-rhev4-2193
    [started]  hax                 0x7200000000000001:0x4f         inet:tcp:cortx-data-headless-svc-ssc-vm-rhev4-2193@22001
    [started]  ioservice           0x7200000000000001:0x52         inet:tcp:cortx-data-headless-svc-ssc-vm-rhev4-2193@21001
    [started]  ioservice           0x7200000000000001:0x5f         inet:tcp:cortx-data-headless-svc-ssc-vm-rhev4-2193@21002
    [started]  confd               0x7200000000000001:0x6c         inet:tcp:cortx-data-headless-svc-ssc-vm-rhev4-2193@21003
    cortx-server-headless-svc-ssc-vm-rhev4-2195
    [started]  hax                 0x7200000000000001:0x71         inet:tcp:cortx-server-headless-svc-ssc-vm-rhev4-2195@22001
    [started]  rgw                 0x7200000000000001:0x74         inet:tcp:cortx-server-headless-svc-ssc-vm-rhev4-2195@21501
    cortx-server-headless-svc-ssc-vm-rhev4-2194
    [started]  hax                 0x7200000000000001:0x79         inet:tcp:cortx-server-headless-svc-ssc-vm-rhev4-2194@22001
    [started]  rgw                 0x7200000000000001:0x7c         inet:tcp:cortx-server-headless-svc-ssc-vm-rhev4-2194@21501
    cortx-server-headless-svc-ssc-vm-rhev4-2193
    [started]  hax                 0x7200000000000001:0x81         inet:tcp:cortx-server-headless-svc-ssc-vm-rhev4-2193@22001
    [started]  rgw                 0x7200000000000001:0x84         inet:tcp:cortx-server-headless-svc-ssc-vm-rhev4-2193@21501
Devices:
    cortx-data-headless-svc-ssc-vm-rhev4-2195
    [online]  /dev/sdd
    [online]  /dev/sde
    [online]  /dev/sdc
    [online]  /dev/sdg
    [online]  /dev/sdh
    [online]  /dev/sdf
    cortx-data-headless-svc-ssc-vm-rhev4-2194
    [online]  /dev/sdg
    [online]  /dev/sdh
    [online]  /dev/sdf
    [online]  /dev/sdc
    [online]  /dev/sdd
    [online]  /dev/sde
    cortx-data-headless-svc-ssc-vm-rhev4-2193
    [online]  /dev/sdd
    [online]  /dev/sde
    [online]  /dev/sdc
    [online]  /dev/sdg
    [online]  /dev/sdh
    [online]  /dev/sdf
    cortx-server-headless-svc-ssc-vm-rhev4-2195
    cortx-server-headless-svc-ssc-vm-rhev4-2194
    cortx-server-headless-svc-ssc-vm-rhev4-2193
```

```
[root@cortx-data-headless-svc-ssc-vm-rhev4-2194 /]# hctl drive-state --json '{"node" : "cortx-data-headless-svc-ssc-vm-rhev4-2194", "source_type" : "drive",  "device" : "/dev/sdh", "state" : "failed"}'
0
2022-03-23 07:20:07,508 [DEBUG] Bound <class 'hax.common.HaxGlobalState'> to an instance <hax.common.HaxGlobalState object at 0x7f30dd1db048>
2022-03-23 07:20:07,509 [DEBUG] Created and configured an injector, config=<function di_configuration at 0x7f30e0930598>
2022-03-23 07:20:07,513 [DEBUG] Starting new HTTP connection (1): 127.0.0.1:8500
2022-03-23 07:20:07,516 [DEBUG] http://127.0.0.1:8500 "GET /v1/kv/eq-epoch HTTP/1.1" 200 97
2022-03-23 07:20:07,524 [DEBUG] http://127.0.0.1:8500 "PUT /v1/txn HTTP/1.1" 200 227
2022-03-23 07:20:07,525 [INFO] Written to epoch: 3
```
```
[root@cortx-data-headless-svc-ssc-vm-rhev4-2194 /]# hctl status -d
Byte_count:
    critical_byte_count : 0
    damaged_byte_count : 0
    degraded_byte_count : 0
    healthy_byte_count : 0
Data pool:
    # fid name
    0x6f00000000000001:0x93 'storage-set-1__sns'
Profile:
    # fid name: pool(s)
    0x7000000000000001:0xd2 'Profile_the_pool': 'storage-set-1__sns' 'storage-set-1__dix' None
Services:
    cortx-data-headless-svc-ssc-vm-rhev4-2195
    [started]  hax                 0x7200000000000001:0x2b         inet:tcp:cortx-data-headless-svc-ssc-vm-rhev4-2195@22001
    [started]  ioservice           0x7200000000000001:0x2e         inet:tcp:cortx-data-headless-svc-ssc-vm-rhev4-2195@21001
    [started]  ioservice           0x7200000000000001:0x3b         inet:tcp:cortx-data-headless-svc-ssc-vm-rhev4-2195@21002
    [started]  confd               0x7200000000000001:0x48         inet:tcp:cortx-data-headless-svc-ssc-vm-rhev4-2195@21003
    cortx-data-headless-svc-ssc-vm-rhev4-2194  (RC)
    [started]  hax                 0x7200000000000001:0x7          inet:tcp:cortx-data-headless-svc-ssc-vm-rhev4-2194@22001
    [started]  ioservice           0x7200000000000001:0xa          inet:tcp:cortx-data-headless-svc-ssc-vm-rhev4-2194@21001
    [started]  ioservice           0x7200000000000001:0x17         inet:tcp:cortx-data-headless-svc-ssc-vm-rhev4-2194@21002
    [started]  confd               0x7200000000000001:0x24         inet:tcp:cortx-data-headless-svc-ssc-vm-rhev4-2194@21003
    cortx-data-headless-svc-ssc-vm-rhev4-2193
    [started]  hax                 0x7200000000000001:0x4f         inet:tcp:cortx-data-headless-svc-ssc-vm-rhev4-2193@22001
    [started]  ioservice           0x7200000000000001:0x52         inet:tcp:cortx-data-headless-svc-ssc-vm-rhev4-2193@21001
    [started]  ioservice           0x7200000000000001:0x5f         inet:tcp:cortx-data-headless-svc-ssc-vm-rhev4-2193@21002
    [started]  confd               0x7200000000000001:0x6c         inet:tcp:cortx-data-headless-svc-ssc-vm-rhev4-2193@21003
    cortx-server-headless-svc-ssc-vm-rhev4-2195
    [started]  hax                 0x7200000000000001:0x71         inet:tcp:cortx-server-headless-svc-ssc-vm-rhev4-2195@22001
    [started]  rgw                 0x7200000000000001:0x74         inet:tcp:cortx-server-headless-svc-ssc-vm-rhev4-2195@21501
    cortx-server-headless-svc-ssc-vm-rhev4-2194
    [started]  hax                 0x7200000000000001:0x79         inet:tcp:cortx-server-headless-svc-ssc-vm-rhev4-2194@22001
    [started]  rgw                 0x7200000000000001:0x7c         inet:tcp:cortx-server-headless-svc-ssc-vm-rhev4-2194@21501
    cortx-server-headless-svc-ssc-vm-rhev4-2193
    [started]  hax                 0x7200000000000001:0x81         inet:tcp:cortx-server-headless-svc-ssc-vm-rhev4-2193@22001
    [started]  rgw                 0x7200000000000001:0x84         inet:tcp:cortx-server-headless-svc-ssc-vm-rhev4-2193@21501
Devices:
    cortx-data-headless-svc-ssc-vm-rhev4-2195
    [online]  /dev/sdd
    [online]  /dev/sde
    [online]  /dev/sdc
    [online]  /dev/sdg
    [online]  /dev/sdh
    [online]  /dev/sdf
    cortx-data-headless-svc-ssc-vm-rhev4-2194
    [online]  /dev/sdg
    [failed]  /dev/sdh
    [online]  /dev/sdf
    [online]  /dev/sdc
    [online]  /dev/sdd
    [online]  /dev/sde
    cortx-data-headless-svc-ssc-vm-rhev4-2193
    [online]  /dev/sdd
    [online]  /dev/sde
    [online]  /dev/sdc
    [online]  /dev/sdg
    [online]  /dev/sdh
    [online]  /dev/sdf
    cortx-server-headless-svc-ssc-vm-rhev4-2195
    cortx-server-headless-svc-ssc-vm-rhev4-2194
    cortx-server-headless-svc-ssc-vm-rhev4-2193
```

**Comment by vaibhavparatwar** [comment]
@pawankumarsrivastava @Pranali04796  please update here on testing from CSM side.

**Comment by pawankumarsrivastava** [comment]
CSM user management working fine. CSM configurations are stored in consul as usual

**Review by pawankumarsrivastava — APPROVED** [review]

**Review by ujjwalpl — APPROVED** [review]

**Comment by jaikumargidwani** [comment]
> @nitin-seagate @madhavemuri @kupranay @sachinpunadikar @cdeshmukh We are planning to update jq and consul package version as part of this change. We have tested image generation and deployment with this change. Refer - [#926 (comment)](https://github.com/Seagate/cortx-re/pull/926#issuecomment-1075474525) Could you please check and confirm this change will not impact the functionality of your component.

@tanujashinde0405 , @kailashyadav11 can you review and update please, thanks

**Comment by tanujashinde0405** [comment]
> > @nitin-seagate @madhavemuri @kupranay @sachinpunadikar @cdeshmukh We are planning to update jq and consul package version as part of this change. We have tested image generation and deployment with this change. Refer - [#926 (comment)](https://github.com/Seagate/cortx-re/pull/926#issuecomment-1075474525) Could you please check and confirm this change will not impact the functionality of your component.
> 
> @tanujashinde0405, @kailashyadav11 can you review and update please, thanks

@shailesh-vaidya, Provisioner doesn't have any dependency on both these 2 packages, We are good from the provisioner side, It won't break anything.

**Comment by cdeshmukh** [comment]
@shailesh-vaidya RGW deployment atleast not affected by these changes

File: docker/cortx-deploy/cortx-control/third-party-rpms.txt
Comment (johnbent) [inline docker/cortx-deploy/cortx-control/third-party-rpms.txt:9]: I like how some lines include a comment showing which repo needs each 3rd party software. Can you add that to these?

**Review by johnbent — APPROVED** [review]

File: docker/cortx-deploy/cortx-control/third-party-rpms.txt
Comment (gauravchaudhari02) [reply docker/cortx-deploy/cortx-control/third-party-rpms.txt:9]: @johnbent, once this PR get merged, we are going to take care of it as another PR.
https://github.com/Seagate/cortx-re/blob/b2a125c0f8ebf3b283930a4d1bdd35ca3124c85e/docker/cortx-deploy/third-party-rpms.txt

**Comment by vaibhavparatwar** [comment]
@shailesh-vaidya @gauravchaudhari02 - can we merge this as we got required approvals? 
@rahul-tripathi-git - Please comment on @sachinpunadikar's question.

**Comment by rahul-tripathi-git** [comment]
For py-utils there is no dependency on both jq and consul versions. 
The Utils ConfStore uses consul as a kv backend and is working as expected.

**Review by rahul-tripathi-git — APPROVED** [review]

**Review by sachinpunadikar — APPROVED** [review]
+1

**Review by yeshpal-jain-seagate — APPROVED** [review]

**Review by shailesh-vaidya — APPROVED** [review]

**Pull request merged**
